src/veh_interact.cpp

```diff
@@ -1353,27 +1353,57 @@ void veh_interact::do_refill()
 
 void veh_interact::calc_overview()
 {
-    const hotkey_queue &hotkeys = hotkey_queue::alphabets();
+    overview_opts.clear();
+    overview_headers.clear();
 
-    const auto next_hotkey = [&]( const vehicle_part & pt, input_event & evt ) {
-        if( overview_action && overview_enable && overview_enable( pt ) ) {
-            const input_event prev = evt;
-            evt = main_context.next_unassigned_hotkey( hotkeys, evt );
-            return prev;
-        } else {
-            return input_event();
+    veh_interact::calc_vehicle_header( veh );
+
+    //call for main vehicle ui
+    overview_opts = get_vehicle_overview( veh, false );
+
+    //group carried vehicle parts by their assigned carry_names stacks top element
+    std::map<std::string, std::vector<vehicle_part>> carried_vehicles;
+
+    for( const vpart_reference &vpr : veh->get_all_parts() ) {
+        if( vpr.part().has_flag( vehicle_part::carried_flag ) ) {
+            carried_vehicles[vpr.part().carry_names.top().substr( vehicle_part::name_offset )].push_back(
+                vpr.part() );
         }
-    };
+    }
 
-    overview_opts.clear();
-    overview_headers.clear();
+    for( std::pair<std::string, std::vector<vehicle_part>> it : carried_vehicles ) {
+        vehicle *tmp_car = new vehicle();
+
+        if( !it.second.empty() && tmp_car->name.empty() ) {
+            tmp_car->name = it.second[0].carried_name();
+        }
+
+        for( vehicle_part &part : it.second ) {
+            tmp_car->install_part( point_zero, part );
+        }
 
-    int epower_w = veh->net_battery_charge_rate_w();
-    overview_headers["1_ENGINE"] = [this]( const catacurses::window & w, int y ) {
+        std::vector<part_option> tmp_overview = get_vehicle_overview( tmp_car, true );
+        overview_opts.insert( overview_opts.end(), tmp_overview.begin(), tmp_overview.end() );
+        free( tmp_car );
```

Comment:
Heap objects allocated by `new` keyword should be freed by `delete` keyword:
```suggestion
        delete tmp_car;
```

But I don't think you really need to allocate a `vehicle` object on heap and release it repeatedly in a loop.

Reply:
ty, should be fixed now (i no longer create vehicles in the loop and just handle it via the parent vehicle)